src/include/txn_inline.h

```diff
@@ -310,26 +311,64 @@ __wt_txn_op_delete_commit_apply_timestamps(WT_SESSION_IMPL *session, WT_REF *ref
      */
     if (previous_state == WT_REF_DELETED) {
         if (ref->ft_info.del->timestamp == WT_TS_NONE) {
-            ref->ft_info.del->timestamp = txn->commit_timestamp;
-            ref->ft_info.del->durable_timestamp = txn->durable_timestamp;
+            if (in_commit_phase) {
```

Comment:
As discussed this logic is a dupe of what we have in `__wt_txn_untimestamped_upd_set_timestamp`, but operating on a `WT_PAGE_DELETED *` instead of a `WT_UPDATE *`. Any recommendations to remove it are welcome.